src/Controllers/DatabaseController.js

```diff
@@ -1570,23 +1570,33 @@ class DatabaseController {
         objectId: userId,
       };
 
-      const ors = permFields.flatMap((key) => {
-        // constraint for single pointer setup
-        const q = {
-          [key]: userPointer,
-        };
-        // constraint for users-array setup
-        const qa = {
-          [key]: { $all: [userPointer] },
-        };
+      const ors = permFields.map((key) => {
+        const fieldDescriptor = schema.getExpectedType(className, key);
+        const fieldType = fieldDescriptor.type;
```

Comment:
The CI tests failed because I can get null/undefined/string in `fieldDescriptor`. Basically I assumed that I could only get a descriptor with a `type` of `Array` or `Pointer`, but it seems this is not the case. I will investigate this issue further and revisit this code. If anyone can point me in the right direction as to why I can get null/undefined or a string as the pointer CLP I will appreciate it.

Reply:
I think I realized that the CI build failed because of the flow type checker and I know why. I am still getting used to flow since I haven't used it before.

Reply:
Fixed it.